### PR TITLE
tests: split CI into unit and smoke workflows

### PR DESCRIPTION
**Pull request by brendankenny**
Splits the smoke stuff out of `ci.yml` and into its own `smoke.yml`. This grew out of frustration with the remaining CI flakes and how "Cancel workflow" -> "Re-run jobs" would then cancel and re-run _all_ of the smoke and unit jobs, when usually it's just a single job. After making the change, though, it's also just easier to read with all the smoke stuff in its own file.

Also tried to improve the CI job names. Check them out down in the checks section. Get your 🚲🏚️s in now :)

### CODE REVIEW

File: .github/workflows/smoke.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+name: smoke
```
Comment (brendankenny) [inline .github/workflows/smoke.yml:1]: nothing changed in here but a few comments and the `name` values for all three jobs

**Comment by connorjclark** [comment]
> Get your 🚲🏚️s in now :)

would love to drop 💡🏠

**Review by connorjclark — APPROVED** [review]

**Comment by brendankenny** [comment]
> > Get your 🚲🏚️s in now :)
> 
> would love to drop 💡🏠

ha, I'm pretty over it too. If you can think of good naming for the workflow and jobs, happy to get a vote going on them :)

**Review by paulirish — APPROVED** [review]
wfm!

**Comment by paulirish** [comment]
@brendankenny you okay owning the "required checks" juggling required for this/ as long as we don't have any land-when-green active this shouldn't be too big a problem.

**Comment by brendankenny** [comment]
> you okay owning the "required checks" juggling required for this?

No problem. I'll try to get it in first thing tomorrow so hopefully it will be relatively painless for everyone.

File: .github/workflows/smoke.yml [hunk 2]
```diff
@@ -19,9 +19,9 @@ jobs:
     runs-on: ubuntu-latest
     env:
       # The smokehouse tests run by job `test set 1`. `test set 2` will run the rest.
-      SMOKE_GROUP_1: a11y oopif pwa pwa2 pwa3 dbw redirects errors offline
```
Comment (connorjclark) [inline .github/workflows/smoke.yml]: `SMOKE_CLOUD_1`? :P

**Comment by connorjclark** [comment]
> ha, I'm pretty over it too. If you can think of good naming for the workflow and jobs, happy to get a vote going on them :)

CI?

**Pull request merged**